common/openapi.yml

```diff
@@ -266,6 +292,31 @@ components:
             $ref: '#/components/schemas/WorkflowOutputDefinition'
           description: Declarative output definition for writing back to WDS.
 
+    RunResultsRequest:
+      type: object
+      required: [ run_id, state ]
+      properties:
+        run_id:
+          type: string
+          format: uuid
+        state:
+          $ref: '#/components/schemas/RunState'
+        outputs:
+          type: object
+          description: Map of output keys to output values
+
+    RunResultsResponse:
+      type: object
+      required: [ run_id, state ]
+      properties:
+        run_id:
+          type: string
+          format: uuid
+        state:
+          $ref: '#/components/schemas/RunState'
+        error:
+          type: string
```

Comment:
I don't think we need an error message here, if there's an error CBAS should return a non-200 code and either a `UserError` or `ServerError` response. In fact, in the case of success I don't think think we need to return any response body at all, just the success code. Realizing this wasn't specified in the design doc, sorry about that!

Reply:
updated!